ios/viewmodels/view-model-motivation.md

```diff
@@ -0,0 +1,109 @@
+
+```
+View logic might be the logic that converts the model type `.HighIntensity` to be represented as a certain color on the screen.
+
+When using views that can display multiple models, you might run into a situation where the view needs to know too much about every kind of model that it can display.
+```swift
+// Don't do this
+
+class ExerciseViewController: UIViewController {
+  @IBOutlet private weak var titleLabel: UILabel!
+
+  func configure(forExercise exercise: Exercise) {
+    let color: UIColor
+    switch excercise.type {
+    case .HighIntensity:
+      color = .red()
+    default:
+      color = .blue()
+    }
+
+    let titleAttributes = [NSForegroundAttributeName: UIFont.cellFont(), NSForegroundAttributeColor: color]
+    let detailAttributes = [NSForegroundAttributeName: UIFont.cellFont(), NSForegroundAttributeColor: UIColor.blackColor()]
+    
+    let durationFormatter = ExerciseDurationFormatter()
+    let title = NSMutableAttributedString(exercise.name + " - ", attributes: titleAttributes)
+    title.appendAttributedString(NSAttributedString(durationFormatter.formatNumber(exercise.duration), attributes: titleAttributes))
+    titleLabel.attributedText = title
+    
+    ...
+    }
+}
+
+```
+
+The view controller here starts to get pretty large just to accomodate the formatting of one label. This problem is compounded for the combination of multiple labels, switches, table views, buttons that are in a view controller.
+
+
+View models can help by moving view logic into the view model class. This simplifies the view implementation by abstracting out the view logic details into a separate class.
+
+```swift
+struct ExerciseViewModel {
+  let title: NSAttributedString
+  
+  init(exercise: Exercise) {
+    let color: UIColor
+    switch excercise.type {
+    case .HighIntensity:
+      color = .red()
+    default:
+      color = .blue()
+    }
+
+    let titleAttributes = [NSForegroundAttributeName: UIFont.cellFont(), NSForegroundAttributeColor: color]
+    let detailAttributes = [NSForegroundAttributeName: UIFont.cellFont(), NSForegroundAttributeColor: UIColor.blackColor()]
+    
+    let durationFormatter = ExerciseDurationFormatter()
+    let title = NSMutableAttributedString(exercise.name + " - ", attributes: titleAttributes)
+    title.appendAttributedString(NSAttributedString(durationFormatter.formatNumber(exercise.duration), attributes: titleAttributes))
+    self.title = title
+  }
+}
+
+```
+
+At first this might seem as though we simply moved the 'mess' from one object to the other, but the responsibilities of these two classes are vastly different. The sole responsibility of ExerciseViewModel is to leave the formatting(in a general sense) of the model out of the view and view controller.
+
+This allows this class to be tested, and for the view to be potentially reused. This also encourages us to think clearly about the models that are required to produce our view, and stating those assumptions explicitly.
```

Comment:
I think it's a good idea but does not need to be in this pr-testing is after all orthogonal to view models.